web/src/message_list_tooltips.ts

```diff
@@ -349,6 +352,65 @@ export function initialize(): void {
         },
     });
 
+    message_list_tooltip(".message_content a[data-message-link-type='external_inline_url']", {
+        onShow(instance) {
+            const $elem = $(instance.reference);
+            const url = $elem.attr("href");
+            instance.setContent(
+                parse_html(render_message_inline_url_tooltip({tooltip_label: url})),
+            );
+        },
+        onHidden(instance) {
+            instance.destroy();
+        },
+        placement: "bottom",
+    });
```

Comment:
I'm not sure I see why you have this separate function with `external_inline_url` as a type -- why can't you just do all the calculations inside the `onShow` function?